scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,31 @@
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+#cython: cdivision=True
+from libc cimport math
+
+
+ctypedef fused real:
+    short
+    int
+    long
+    float
+    double
```

Comment:
Moreover, one needs to check that `data1.dtype == data2.dtype` before calling this function.